hie-plugin-api/Haskell/Ide/Engine/PluginDescriptor.hs

```diff
@@ -214,7 +233,8 @@ data ParamVal (t :: ParamType) where
 class ValidResponse a where
   jsWrite :: a -> Object -- ^ Serialize to JSON Object
   jsRead  :: Object -> Parser a -- ^ Read from JSON Object
-
+  responseTypeName :: a -> T.Text -- ^ Type name in command description.
```

Comment:
I wonder if this should not be the `TypeRep` of the return type, so it actually tags the return type?

This means we are stuck with long names in the serialisation, but at least we get a guarantee about the tag matching the type.